### PR TITLE
CI: install tree-sitter-cli from npm

### PR DESCRIPTION
**Pull request by stsewd**
This should fix the CI on https://github.com/nvim-treesitter/nvim-treesitter/pull/1645

### CODE REVIEW

File: .github/workflows/check-query-files-and-compilation.yml [hunk 1]
```diff
@@ -33,6 +33,9 @@ jobs:
         run: |
           bash ./scripts/ci-install-${{ matrix.os }}.sh
 
+      - name: Install tree-sitter-cli
+        run: npm install -g tree-sitter-cli
```
Comment (stsewd) [inline .github/workflows/check-query-files-and-compilation.yml:37]: We can also pin this to the current version, but maybe is good to help testing new releases?

**Comment by stsewd** [comment]
> Warning: parser for haskell is not installed

Only haskell is skipped now, yay!

**Comment by stsewd** [comment]
> Failed to load parser: uv_dlopen: A dynamic link library (DLL) initialization routine failed.^M

No idea why windows is failing for some parsers now... they aren't even required to install from grammar, so nothing should have changed for those...

**Review by theHamsta — APPROVED** [review]

File: .github/workflows/check-query-files-and-compilation.yml (hunk 1 above)
Comment (theHamsta) [reply .github/workflows/check-query-files-and-compilation.yml:37]: better stay up-to-date

**Comment by theHamsta** [comment]
It does not work on Windows

**Pull request closed without merging**